src/Windowing/Silk.NET.Windowing.Common/WindowExtensions.cs

```diff
@@ -43,21 +43,29 @@ public static void Invoke(this IView view, Action t)
         public static void Run(this IView view)
         {
             view.Initialize();
-            view.Run
-            (
-                () =>
+
+            void OnFrame()
+            {
+                view.DoEvents();
+                DoFrame();
+            }
+
+            void DoFrame()
+            {
+                if (!view.IsClosing)
                 {
-                    view.DoEvents();
-                    if (!view.IsClosing)
-                    {
-                        view.DoUpdate();
-                    }
-                    if (!view.IsClosing)
-                    {
-                        view.DoRender();
-                    }
+                    view.DoUpdate();
                 }
-            );
+
+                if (!view.IsClosing)
+                {
+                    view.DoRender();
+                }
+            }
+
+            view.Refresh += DoFrame;
```

Comment:
I appreciate the intent, but I don't think automatically binding to events is good. If we want this to be default behaviour, then maybe we should just make it so. Wouldn't be the worst idea actually, then we wouldn't need Working Group approval at all and we can sweep the SDL vs GLFW behaviour differences under the rug as "undefined behaviour" (seeing as we only officially support SDL on Android and experimentally on iOS)